src/crypto/siphash.cuh

```diff
@@ -63,7 +63,7 @@ public:
     v0 = vectorize(sk.k0); v1 = vectorize(sk.k1); v2 = vectorize(sk.k2); v3 = vectorize(sk.k3);
   }
   __device__ uint64_t xor_lanes() {
-    return devectorize((v0 ^ v1) ^ (v2  ^ v3));
+    return devectorize(rotl((v0 ^ v1) ^ (v2  ^ v3), 17));
```

Comment:
Compared to upstream, this is the only line changed in ` src/crypto/siphash.cuh` compared to the two that there were before. That makes sense as upstream factored. Ok.